### ISSUE TITLE
[Question] Nomad task bind to different ports on a host

### ISSUE THREAD

**Issue by achilu** [hashicorp/nomad#6264]
I have a question with Nomad hcl, how can we bind 3 or x static ports on a host for a single task, will the scheduling algorithm be able to pick the port that is left unused when the task is deployed on the host.
 
Say for example:
 
I have 443, 444, 445 declared as static ports to be used by a job, when nomad place’s the task on host A , the 1st allocation of the task should use 443 or 444 or 445 (let’s assume it used 443), 2nd allocation should pick from 444 or 445 .. and so on.
 
So each of the task which runs a docker container will run the service inside the container on port 443, which will be mapped to 443 or 444 or 445 on the host?

In short I am looking to see if there is a way to define a port constraint on the nomad job, that will place the allocations on a host that has one of the port's available. 
 
Please let me know if this doable?

**Comment by achilu** [comment]
can you please take a look and comment on this?

**Comment by notnoop** [comment]
Thanks @achilu for reaching out.  We don't support such pattern now and this seems like a large departure of how we express constraints, so it's unlikely for us to support.

That being said, mind if you provide some context on constraints forcing you to only use 3 ports but not others?

Probably another way of phrasing the question is: how to configure a specific range of dynamic ports so an alloc is assigned a port from a small range rather than the client large range of dynamic ports?
